lib/requestwrapper.ts

```diff
@@ -192,6 +196,7 @@ export function createRequest(parameters, _callback) {
     // Remove non-valid inputs for buildRequestFileObject,
     // i.e things like {contentType: <contentType>}
     Object.keys(formData).forEach(key => {
+      // tslint:disable-next-line:no-unused-expression
```

Comment:
Why don't we make the logic easier to read and refactor this block that parses the `formData` into it's function to make testing easier? 

Something like this maybe

```
 function parseFormData(formData: Object): Object {
   let _formData = extend({}, formData);
   // cleanup file formData with null/undefined data
   Object.keys(_formData).forEach(key => {
     if (_formData[key].data == null) {
       delete _formData[key];
     }
   });
   // cleanup empty objects
   Object.keys(_formData).forEach(key => {
     if (isEmptyObject(_formData[key])) {
       delete _formData[key];
     }
   });
   // convert file formData parameters to request-style objects
   // and stringify object formData parameters
   Object.keys(_formData).forEach(key => {
     if (isFileParam(_formData[key])) {
       _formData[key] = buildRequestFileObject(_formData[key]);
     } else if (typeof formData[key] === 'object') {
       _formData[key] = JSON.stringify(_formData[key]);
     }
   });
   return _formData;
 }
```